python/pyspark/mllib/linalg/distributed.py

```diff
@@ -0,0 +1,513 @@
+        rows = rows.map(_convert_to_vector)
+
+        if not (isinstance(java_matrix, JavaObject)
+                and java_matrix.getClass().getSimpleName() == "RowMatrix"):
+            java_matrix = callMLlibFunc("createRowMatrix", rows, long(numRows), int(numCols))
+
+        self._java_matrix_wrapper = JavaModelWrapper(java_matrix)
+        self._rows = rows
+
+    @property
+    def rows(self):
+        """Rows of the RowMatrix stored as an RDD of vectors."""
+        return self._rows
+
+    def numRows(self):
+        """
+        Get or compute the number of rows.
+
+        >>> rows = sc.parallelize([[1, 2, 3], [4, 5, 6],
+        ...                        [7, 8, 9], [10, 11, 12]])
+
+        >>> rm = RowMatrix(rows)
+        >>> print(rm.numRows())
+        4
+
+        >>> rm = RowMatrix(rows, 7, 6)
+        >>> print(rm.numRows())
+        7
+        """
+        return self._java_matrix_wrapper.call("numRows")
+
+    def numCols(self):
+        """
+        Get or compute the number of cols.
+
+        >>> rows = sc.parallelize([[1, 2, 3], [4, 5, 6],
+        ...                        [7, 8, 9], [10, 11, 12]])
+
+        >>> rm = RowMatrix(rows)
+        >>> print(rm.numCols())
+        3
+
+        >>> rm = RowMatrix(rows, 7, 6)
+        >>> print(rm.numCols())
+        6
+        """
+        return self._java_matrix_wrapper.call("numCols")
+
+
+class IndexedRow(object):
+    """
+    .. note:: Experimental
+
+    Represents a row of an IndexedRowMatrix.
+
+    Just a wrapper over a (long, vector) tuple.
+
+    :param index: The index for the given row.
+    :param vector: The row in the matrix at the given index.
+    """
+    def __init__(self, index, vector):
+        self.index = long(index)
+        self.vector = _convert_to_vector(vector)
+
+    def __repr__(self):
+        return "IndexedRow(%s, %s)" % (self.index, self.vector)
+
+
+def _convert_to_indexed_row(row):
+    if isinstance(row, IndexedRow):
+        return row
+    elif isinstance(row, tuple) and len(row) == 2:
+        return IndexedRow(*row)
+    else:
+        raise TypeError("Cannot convert type %s into IndexedRow" % type(row))
+
+
+class IndexedRowMatrix(DistributedMatrix):
+    """
+    .. note:: Experimental
+
+    Represents a row-oriented distributed Matrix with indexed rows.
+
+    :param rows: An RDD of IndexedRows or (long, vector) tuples.
+    :param numRows: Number of rows in the matrix. A non-positive
+                    value means unknown, at which point the number
+                    of rows will be determined by the max row
+                    index plus one.
+    :param numCols: Number of columns in the matrix. A non-positive
+                    value means unknown, at which point the number
+                    of columns will be determined by the size of
+                    the first row.
+    """
+    def __init__(self, rows, numRows=0, numCols=0, java_matrix=None):
+        """Create a wrapper over a Java IndexedRowMatrix."""
```

Comment:
It is okay to use docstring here. But the problem is the content `Create a wrapper over a Java IndexedRowMatrix` is not sufficient for the `__init__` doc. In the generated Python doc, the docstring for the class is used for the constructor rather than this.